modules/manage/pages/kubernetes/k-rolling-restart.adoc

```diff
@@ -0,0 +1,107 @@
+= Perform a Rolling Restart of Redpanda in Kubernetes
+:description: Learn how to perform a rolling restart of your Redpanda cluster when it's running in Kubernetes.
+:rolling-restart:
+:page-context-links: [{"name": "Linux", "to": "manage:cluster-maintenance/rolling-restart.adoc" },{"name": "Kubernetes", "to": "manage:kubernetes/k-rolling-restart.adoc" } ]
+
+A rolling restart involves restarting one broker at a time, while the remaining brokers in your cluster continue running. This is to minimize downtime during a full cluster restart.
+Perform a rolling restart during operations such as configuration updates that require a restart, version upgrades or cluster maintenance.
+
+When you run Redpanda in Kubernetes with the Redpanda Operator or the Redpanda Helm chart, your Redpanda cluster is managed as a StatefulSet where each broker runs inside its own Pod. As a result, you can perform a rolling restart using a single command with the Kubernetes API.
+
+== Prerequisites
+
+You must have the following:
+
+* xref:deploy:deployment-option/self-hosted/kubernetes/index.adoc[A Redpanda cluster running in Kubernetes].
+
+include::upgrade:partial$rolling-upgrades/restart-impact.adoc[leveloffset=+1]
+
+== Perform a rolling restart
+
+A rolling restart involves running the following procedure on each broker, one at a time. The Redpanda Operator and the Redpanda Helm chart automate this procedure.
+
+. Place a broker into maintenance mode.
+. Wait for maintenance mode to finish.
+. Terminate the Pod so that it is restarted.
+. Take the broker out of maintenance mode.
+
+When a broker is placed into maintenance mode, it reassigns its partition leadership to other brokers for all topics that have a replication factor greater than one (three is the default replication factor for topics). Reassigning partition leadership involves _draining_ leadership from the broker and _transferring_ that leadership to another broker.
+
+[NOTE]
+====
+The Redpanda Helm chart's `preStop` lifecycle hook puts each broker into maintenance mode before the Pod that it's running on is terminated. If maintenance mode does not complete before the `terminationGracePeriod` the container is forcefully terminated using a `SIGKILL` command.
+
+The default `terminationGracePeriod` is 90 seconds, which should be long enough for large clusters.
+You can test different values in a development environment.
+To configure the `terminationGracePeriod`,
+use the xref:reference:redpanda-helm-spec.adoc#statefulsetterminationgraceperiodseconds[`statefulset.terminationGracePeriodSeconds`] setting.
+====
+
+. Check for topics that have a replication factor greater than one.
```

Comment:
I feel like we're missing a subheading before this list. I'm not quite following the flow of the guide here. We list the four top-level steps and read about going into maintenance mode. I'm not quite clear where these steps fall within the above list, if they do at all?